pkg/operator/operator.go

```diff
@@ -77,6 +85,12 @@ func NewOperator(ctx context.Context, operator *operator.Operator) (context.Cont
 		options.FromContext(ctx).VnetGUID = vnetGUID
 	}
 
+	// These options are set similarly to those used by operator.KubernetesInterface
+	inClusterConfig := lo.Must(rest.InClusterConfig())
+	inClusterConfig.RateLimiter = flowcontrol.NewTokenBucketRateLimiter(float32(coreoptions.FromContext(ctx).KubeClientQPS), coreoptions.FromContext(ctx).KubeClientBurst)
+	inClusterConfig.UserAgent = "karpenter"
```

Comment:
Could use `GetUserAgentExtension` (which should probably be called GetUserAgent, and is being fixed in #819)

Reply:
Is that what we want for the kube-client useragent too? It's not what upstream Karpenter is currently using for its kube agent, it just uses `karpenter`

Reply:
I think so. I can't think of a very good reason to omit the version. I can think of some (maybe ease of querying if one can't do partial matches, or to minimize request payload, etc.) - but they don't seem to be very convincing. I say we use the same - until we find out why this is wrong :-)